From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Samuel Maddock <smaddock@slack-corp.com>
Date: Wed, 8 Oct 2025 10:50:03 -0400
Subject: fix: js2c including mac sdk headers twice

https://chromium-review.googlesource.com/c/chromium/src/+/6950738

Chromium has set the minimum MacOS SDK version to 26. With this
change, it seems to introduce an incompatibility when compiling
using clang modules. Disabling them seems to resolve the issue.

diff --git a/unofficial.gni b/unofficial.gni
index a64d2e4ac475abc049fff7ea62ec76de565a747d..61aeac679a90832a1d9f559a4eb880362bce038d 100644
--- a/unofficial.gni
+++ b/unofficial.gni
@@ -350,6 +350,12 @@ template("node_gn_build") {
     include_dirs = [ "src", "tools" ]
     configs += [ "//build/config/compiler:no_exit_time_destructors" ]
 
+    # Fix for MacOSX26 SDK headers included twice due to usage of clang modules.
+    # Included once as a C header and again as C++.
+    if (is_mac) {
+      use_libcxx_modules = false
+    }
+
     if (!is_win) {
       defines += [ "NODE_JS2C_USE_STRING_LITERALS" ]
     }
